innkeeper/client.go

```diff
@@ -98,67 +91,200 @@ type apiError struct {
 	ErrorType string `json:"type"`
 }
 
-type Client struct {
-	baseUrl          *url.URL
-	auth             Authentication
-	authToken        string
-	lastModified     string
-	preRouteFilters  []string
-	postRouteFilters []string
-	httpClient       *http.Client
-	dataChan         chan string
-	routeCache       routeCache
-	closer           chan interface{}
+// Provides an Authentication implementation with fixed token string
+type FixedToken string
+
+func (fa FixedToken) Token() (string, error) { return string(fa), nil }
+
+type Authentication interface {
+	Token() (string, error)
 }
 
 type Options struct {
 	Address          string
 	Insecure         bool
-	PollTimeout      time.Duration
 	Authentication   Authentication
-	PreRouteFilters  []string
-	PostRouteFilters []string
+	PreRouteFilters  string
+	PostRouteFilters string
+}
+
+type Client struct {
+	opts             Options
+	preRouteFilters  []*eskip.Filter
+	postRouteFilters []*eskip.Filter
+	authToken        string
+	httpClient       *http.Client
+	lastChanged      string
 }
 
-// Creates an innkeeper client.
 func New(o Options) (*Client, error) {
-	u, err := url.Parse(o.Address)
+	preFilters, err := eskip.ParseFilters(o.PreRouteFilters)
 	if err != nil {
 		return nil, err
 	}
 
-	c := &Client{
-		u, o.Authentication, "", "",
-		o.PreRouteFilters, o.PostRouteFilters,
-		&http.Client{Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: o.Insecure}}},
-		make(chan string),
-		make(routeCache),
-		make(chan interface{})}
-
-	// start a polling loop
-	go func() {
-		to := time.Duration(0)
-		for {
-			select {
-			case <-time.After(to):
-				to = o.PollTimeout
-				if c.poll() {
-					c.dataChan <- toDocument(c.routeCache)
-				}
-			case <-c.closer:
-				return
-			}
+	postFilters, err := eskip.ParseFilters(o.PostRouteFilters)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		opts:             o,
+		preRouteFilters:  preFilters,
+		postRouteFilters: postFilters,
+		httpClient: &http.Client{Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: o.Insecure}}}}, nil
+}
+
+func splitOnMethods(data []*routeData) []*routeData {
+	var split []*routeData
+	for _, d := range data {
+		if len(d.Route.MatchMethods) == 0 {
+			split = append(split, d)
+			continue
+		}
+
+		for _, m := range d.Route.MatchMethods {
+			copy := d.Route
+			copy.matchMethod = m
+			split = append(split, &routeData{d.Id, d.CreatedAt, d.DeletedAt, copy})
+		}
+	}
+
+	return split
+}
+
+func convertHeaders(d *routeData) map[string]string {
+	hs := make(map[string]string)
+	for _, h := range d.Route.MatchHeaders {
+		hs[h.Name] = h.Value
+	}
+
+	return hs
+}
+
+func innkeeperProcotolToScheme(p string) string {
+	return strings.ToLower(p)
```

Comment:
right, ok (I wish innkeeper just sent the actual value, since it is already a formal value, no need for a redefined enumeration)